3cqsbot.py

```diff
@@ -313,16 +313,25 @@ async def botswitch():
 async def dcaconfswitch():
     
     while True:
-        if asyncState.fgi >= attributes.get("fgi_min","","fgi_defensive") and asyncState.fgi <= attributes.get("fgi_max","","fgi_defensive"):
+        if asyncState.fgi >= attributes.get("fgi_min", 0, "fgi_defensive") and asyncState.fgi <= attributes.get("fgi_max", 30, "fgi_defensive"):
             asyncState.dca_conf = "fgi_defensive"
 
-        if asyncState.fgi >= attributes.get("fgi_min","","fgi_moderate") and asyncState.fgi <= attributes.get("fgi_max","","fgi_moderate"):
+        if asyncState.fgi >= attributes.get("fgi_min", 31, "fgi_moderate") and asyncState.fgi <= attributes.get("fgi_max", 60, "fgi_moderate"):
             asyncState.dca_conf = "fgi_moderate"
 
-        if asyncState.fgi >= attributes.get("fgi_min","","fgi_aggressive") and asyncState.fgi <= attributes.get("fgi_max","","fgi_aggressive"):
+        if asyncState.fgi >= attributes.get("fgi_min", 61, "fgi_aggressive") and asyncState.fgi <= attributes.get("fgi_max", 100, "fgi_aggressive"):
             asyncState.dca_conf = "fgi_aggressive" 
 
-        if attributes.get("single","",asyncState.dca_conf):
```

Comment:
This block should be after while True (line 316) and before the dca_conf checks, shouldn't it?

Reply:
moved